src/test/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessorIT.java

```diff
@@ -252,6 +257,58 @@ private void createTextEmbeddingIndex() throws Exception {
         );
     }
 
+    public void testAsymmetricTextEmbeddingProcessor() throws Exception {
+        String modelId = null;
+        try {
+            modelId = uploadAsymmetricEmbeddingModel();
+            loadModel(modelId);
+            createPipelineProcessor(modelId, PIPELINE_NAME, ProcessorType.TEXT_EMBEDDING, 2);
+            createTextEmbeddingIndex();
+            ingestDocument();
+            assertEquals(1, getDocCount(INDEX_NAME));
+        } finally {
+            wipeOfTestResources(INDEX_NAME, PIPELINE_NAME, modelId, null);
+        }
+    }
+
+    private void ingestDocument() throws Exception {
```

Comment:
Is it possible to reuse the below `ingestDocument(String, String)`? Or at least we can put this document to a json file for better readability and maintenance, WDYT?

Reply:
Yes, sure. Good catch.